subprojects/docs/src/samples/workerApi/noIsolation/kotlin/build.gradle.kts

```diff
@@ -0,0 +1,41 @@
+// tag::unit-of-work[]
+
+import org.gradle.workers.WorkerExecutor
+
+import javax.inject.Inject
+
+// The implementation of a single unit of work
+open class ReverseFile @Inject constructor(val fileToReverse: File, val destinationFile: File) : Runnable {
+
+    override fun run() {
+        destinationFile.writeText(fileToReverse.readText().reversed())
+    }
+}
+// end::unit-of-work[]
+
+// tag::task-implementation[]
+// The WorkerExecutor will be injected by Gradle at runtime
+open class ReverseFiles @Inject constructor(val workerExecutor: WorkerExecutor) : SourceTask() {
+    @OutputDirectory
+    var outputDir: File? = null
+
+    @TaskAction
+    fun reverseFiles() {
+        // Create and submit a unit of work for each file
+        getSource().forEach { file ->
```

Comment:
Note: `getSource()` is used because `source` is resolved to the protected field of type List, which contains the string "source" stored there by the configuration.

Reply:
Good point! IDEA is telling me that this field isn't used in any of the `SourceTask` descendants. Could you open an issue on `gradle/gradle` about that protected field hiding the Kotlin property defined by the corresponding getter and setter?

Reply:
Done: #6641

Reply:
Follow up fix in https://github.com/gradle/gradle/pull/6850